In [1]:
import pandas as pd
data = pd.read_csv('thanksgiving.csv',encoding='Latin-1')
print(data.head(5))

   RespondentID Do you celebrate Thanksgiving?  \
0    4337954960                            Yes   
1    4337951949                            Yes   
2    4337935621                            Yes   
3    4337933040                            Yes   
4    4337931983                            Yes   

  What is typically the main dish at your Thanksgiving dinner?  \
0                                             Turkey             
1                                             Turkey             
2                                             Turkey             
3                                             Turkey             
4                                           Tofurkey             

  What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
0                                                NaN                                      
1                                                NaN                                      
2                            

In [2]:
print(data.columns)

Index(['RespondentID', 'Do you celebrate Thanksgiving?',
       'What is typically the main dish at your Thanksgiving dinner?',
       'What is typically the main dish at your Thanksgiving dinner? - Other (please specify)',
       'How is the main dish typically cooked?',
       'How is the main dish typically cooked? - Other (please specify)',
       'What kind of stuffing/dressing do you typically have?',
       'What kind of stuffing/dressing do you typically have? - Other (please specify)',
       'What type of cranberry saucedo you typically have?',
       'What type of cranberry saucedo you typically have? - Other (please specify)',
       'Do you typically have gravy?',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots',
       'Which of these side dishes aretypically served

In [3]:
data['Do you celebrate Thanksgiving?'].value_counts()

Yes    980
No      78
Name: Do you celebrate Thanksgiving?, dtype: int64

In [4]:
filter = data['Do you celebrate Thanksgiving?'] == 'Yes'
data = data[filter]
data['Do you celebrate Thanksgiving?'].value_counts()

Yes    980
Name: Do you celebrate Thanksgiving?, dtype: int64

In [5]:
data['What is typically the main dish at your Thanksgiving dinner?'].value_counts()

Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
I don't know                5
Turducken                   3
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64

In [6]:
data[data['What is typically the main dish at your Thanksgiving dinner?']=='Tofurkey']['Do you typically have gravy?']

4      Yes
33     Yes
69      No
72      No
77     Yes
145    Yes
175    Yes
218     No
243    Yes
275     No
393    Yes
399    Yes
571    Yes
594    Yes
628     No
774     No
820     No
837    Yes
860     No
953    Yes
Name: Do you typically have gravy?, dtype: object

In [7]:
apple_isnull = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple'].isnull()
pumpkin_isnull = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin'].isnull()
pecan_isnull = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan'].isnull()
ate_pies = apple_isnull & pumpkin_isnull & pecan_isnull
ate_pies.value_counts()

False    876
True     104
dtype: int64

In [14]:
def get_age(string):
    if type(string) == str:
        split = string.split(' ')
        split[0] = split[0].replace('+','')
        x = int(split[0])
        return x
    else:
        return None

In [15]:
data['int_age'] = data['Age'].apply(get_age)
data['int_age'].describe()

count    947.000000
mean      40.089757
std       15.352014
min       18.000000
25%       30.000000
50%       45.000000
75%       60.000000
max       60.000000
Name: int_age, dtype: float64

# Finding So Far - 1
## Surveyed Behaviors
Overwhelming percentage of the survey participants celebrated Thanksgiving (980/1058, 92.6%). Among them, 87.7% (859/980) had Turkey as the main dish. We also noted that their appetites were not satisfied by the main dish alone, as 89.4% (876/980) of them also had dessert dishes of either apple, pumpkin, or pecan flavors.

## Demographics
Of the survey participants who celebrated Thanksgiving, 96.7% (947/980) of them had provided information regarding their age. It showed that their average age is approximately 40 years old, ranging from 16 to 60 years old. However, it should be noted that our approach in deducing this data may be biased: the original data was of categorical nature; however, in order to compute descriptive statistics, we took the first age number that showed up in the age categories as the participants' ages.

In [16]:
def get_income(data):
    if pd.isnull(data):
        return None
    else:
        split = data.split(' ')
        if split[0] == 'Prefer':
            return None
        else:
            split[0] = split[0].replace('$','')
            split[0] = split[0].replace(',','')
            x = int(split[0])
            return x

In [18]:
data['int_income'] = data['How much total combined money did all members of your HOUSEHOLD earn last year?'].apply(get_income)
data['int_income'].describe()

count       829.000000
mean      75965.018094
std       59068.636748
min           0.000000
25%       25000.000000
50%       75000.000000
75%      100000.000000
max      200000.000000
Name: int_income, dtype: float64

# Finding So Far - 2
## Household Income
Among the participants who have celebrated Thanksgiving, 84.6% (829/980) of them have provided information regarding their annual household income. Their average being \$75965/year, ranging from \$0/year to \$2000000/year. However, since we used the same approach that was used in extracting age data, we should note that this finding is inaccurate. In fact, since we used the floor of each category to represent each point of data, our finding would be under-evaluated when compared to the reality.

In [28]:
filter = data['int_income'] < 150000
data_low_inc = data[filter]

In [35]:
data_low_inc['How far will you travel for Thanksgiving?'].value_counts() / data_low_inc['How far will you travel for Thanksgiving?'].value_counts().sum()

Thanksgiving is happening at my home--I won't travel at all                         0.407837
Thanksgiving is local--it will take place in the town I live in                     0.294630
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    0.217707
Thanksgiving is out of town and far away--I have to drive several hours or fly      0.079826
Name: How far will you travel for Thanksgiving?, dtype: float64

In [34]:
filter = data['int_income'] > 150000
data_high_inc = data[filter]
data_high_inc['How far will you travel for Thanksgiving?'].value_counts()
data_high_inc['How far will you travel for Thanksgiving?'].value_counts() / data_high_inc['How far will you travel for Thanksgiving?'].value_counts().sum()

Thanksgiving is happening at my home--I won't travel at all                         0.480392
Thanksgiving is local--it will take place in the town I live in                     0.245098
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    0.156863
Thanksgiving is out of town and far away--I have to drive several hours or fly      0.117647
Name: How far will you travel for Thanksgiving?, dtype: float64

# Finding So Far - 3
## Hypothesis: People earning less money could be younger, and would travel to their parent's houses for Thanksgiving. People earning more are more likely to have Thanksgiving at their house as a result.
It is postulated that there is a positive correlation between age and income: younger people make less while the older make more. It is further postulated that younger people would travel to their parents' houses for the Thanksgiving while the older crowd would stay at their own homes. In conjecture, the two postulations imply that people who make less money would travel farther while the people who make more would stay at home.

To verify this hypothesis, we separated the data into two groups based on the previously extracted annual household income, with \$150000/year being the dividing line. We then investigated the categorical distributions for the question: How far will you travel for Thanksgiving?

According to our finding, the lower income group had a distribution of 40.1%, 29.5%, 21.8%, and 8.0% for the four categorical groups. On the other hand, the higher income group had a distrubtion of 48.0%, 24.5%, 15.7%, and 11.8%. By the glance, the distinctions between these two groups was not apparent, though should we desire, we can apply statistical methods to test their significances.

In [40]:
data.pivot_table(index='Have you ever tried to meet up with hometown friends on Thanksgiving night?',columns='Have you ever attended a "Friendsgiving?"',values='int_age')

Have you ever attended a "Friendsgiving?"                  No        Yes
Have you ever tried to meet up with hometown fr...                      
No                                                  42.283702  37.010526
Yes                                                 41.475410  33.976744

In [41]:
data.pivot_table(index='Have you ever tried to meet up with hometown friends on Thanksgiving night?',columns='Have you ever attended a "Friendsgiving?"',values='int_income')

Have you ever attended a "Friendsgiving?"                     No           Yes
Have you ever tried to meet up with hometown fr...                            
No                                                  78914.549654  72894.736842
Yes                                                 78750.000000  66019.736842

# Finding So Far - 4
## Hypothesis: "Friendsgiving" is when instead of traveling home for the holiday, you celebrate it with friends who live in your area. Both questions seem skewed towards younger people.
We used two pivot tables to compare the average age and average income among 4 groups of participants: people who answered Yes to both questions, answered Yes to one question, and so on.

The first pivot table, showing the average age, indicated that people who have tried and attended Friendsgiving were indeed of younger age.

The second pivot table, showing the average income, indicated that people who have tried and attended Friendsgiving were indeed of lower income.

With this approach, we identified that there was a positive correlation between age and income.